client/packages/system/src/Item/Components/StockItemSearchInput/StockItemSearchInput.tsx

```diff
@@ -19,25 +24,50 @@ export const StockItemSearchInput: FC<StockItemSearchInputProps> = ({
   autoFocus = false,
   openOnFocus,
 }) => {
-  const { data, isLoading } = useItemStockOnHand();
+  const [items, setItems] = useState<ItemStockOnHandFragment[]>([]);
+  const { pagination, onPageChange } = usePagination();
+  const { filter, onFilter } = useOrganisationFilter();
+
+  const { data, isLoading } = useItemStockOnHand({
+    pagination,
+    filter,
+  });
   const t = useTranslation('common');
   const formatNumber = useFormatNumber();
 
-  const value = data?.nodes.find(({ id }) => id === currentItemId) ?? null;
+  const value = items.find(({ id }) => id === currentItemId) ?? null;
   const selectControl = useToggle();
 
-  const options = extraFilter
-    ? data?.nodes?.filter(extraFilter) ?? []
-    : data?.nodes ?? [];
+  const options = extraFilter ? items.filter(extraFilter) ?? [] : items ?? [];
```

Comment:
doing this change also removes a few extra renders (and makes the options prop below simple (`options={options}`)
```suggestion
const options = React.useMemo(
    () =>
      defaultOptionMapper(
        extraFilter ? items.filter(extraFilter) ?? [] : items ?? [],
        'name'
      ),
    [items]
  );
```

Reply:
Oops, sorry I missed this part to update. Here 7bb6a3dfb